e2e/cypress/integration/system_console/settings_spec.js

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod
+// Group: @system_console
+
+import ldapUsers from '../../fixtures/ldap_users.json';
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+const authenticator = require('authenticator');
+
+describe('Settings', () => {
+    let mfaUser;
+    let samlUser;
+    let testSettings;
+
+    const ldapUser = ldapUsers['test-1'];
+
+    before(() => {
+        cy.apiInitSetup().
+            then(({user}) => {
+                mfaUser = user;
+
+                cy.apiUpdateConfig({
+                    ServiceSettings: {
+                        EnableMultifactorAuthentication: true,
+                    },
+                });
+
+                // * Check if server has license for LDAP
+                cy.apiRequireLicenseForFeature('LDAP');
+
+                // # Test LDAP configuration and server connection
+                // # Synchronize user attributes
+                cy.apiLDAPTest();
+                cy.apiLDAPSync();
+
+                return cy.apiGetConfig();
+            }).then(({config}) => {
+                testSettings = setLDAPTestSettings(config);
+                testSettings.user = ldapUser;
+                return cy.doLDAPLogin(testSettings);
+            }).then(() => {
+                return cy.apiLogout();
+            }).then(() => {
+                return cy.apiAdminLogin();
+            }).then(() => {
+                cy.visit('/admin_console');
+                return cy.apiCreateUser();
+            }).then(({user: user2}) => {
+                // # Create SAML user
+                samlUser = user2;
+                const body = {
+                    from: 'email',
+                    auto: false,
+                };
+                body.matches = {};
+                body.matches[user2.email] = user2.username;
+
+                return migrateAuthToSAML(body);
+            }).then(() => {
+                return cy.apiGenerateMfaSecret(mfaUser.id);
+            }).then((res) => {
+                // # Create MFA user
+                const token = authenticator.generateToken(res.code.secret);
+
+                return cy.apiActivateUserMFA(mfaUser.id, true, token);
+            });
+    });
+
+    it('MM-T953: Verify correct authentication method', () => {
```

Comment:
Please extract this into its own separate spec file at `/enterprise` folder and add the following test metadata:
```
// Group: @enterprise @system_console @mfa
```

This test will most likely run in a separate test groups together with other MFA and similar test.  Thanks.